physics/SFC_Models/Land/sfc_land.F90

```diff
@@ -75,32 +95,76 @@ subroutine sfc_land_run(im, cpllnd, cpllnd2atm, flag_iter, dry,   &
    character(len=*)    , intent(out)   :: errmsg
    integer             , intent(out)   :: errflg
 
+   ! Constant parameters
+   real(kind=kind_phys), parameter :: &
+  &    one  = 1.0_kind_phys, &
+  &    zero = 0.0_kind_phys, &
+  &    qmin = 1.0e-8_kind_phys
+
    ! Locals
    integer :: i
+   real(kind=kind_phys) :: qss, rch, tem, cpinv, hvapi, elocp
+   real(kind=kind_phys), dimension(im) :: rho, q0
 
    ! Initialize CCPP error handling variables
    errmsg = ''
    errflg = 0
 
+   cpinv = one/cp
+   hvapi = one/hvap
+   elocp = hvap/cp
+
    ! Check coupling from component land to atmosphere
    if (.not. cpllnd2atm) return
 
-   ! Fill variables
-   do i = 1, im
-      sncovr1(i) = sncovr1_lnd(i)
-      qsurf(i)   = qsurf_lnd(i)
-      hflx(i)    = hflx_lnd(i)
-      evap(i)    = evap_lnd(i)
-      ep(i)      = ep_lnd(i)
-      t2mmp(i)   = t2mmp_lnd(i)
-      q2mp(i)    = q2mp_lnd(i)
-      gflux(i)   = gflux_lnd(i)
-      drain(i)   = drain_lnd(i)
-      runoff(i)  = runoff_lnd(i)
-      cmm(i)     = cmm_lnd(i)
-      chh(i)     = chh_lnd(i)
-      zvfun(i)   = zvfun_lnd(i)
-   enddo
+   ! Check if it is cold or warm run
+   if (flag_init .and. .not. flag_restart) then 
```

Comment:
@barlage I add this part to calculate land fluxes in the first time step. Let me know if you see any issue with it.

Reply:
this makes sense to me

Reply:
@barlage The part that calculated land fluxes internally is producing very large values for latent and sensible heat flux. I am trying to debug it but it might be related to the way of calculation of qss (not sure now). Since this is based on sfc_ocean it might need to have some adjustment. Any suggestion?

Reply:
@uturuncoglu I was hoping we could get away with this simple approximation but perhaps there needs to be a constraint added based on available energy. Typically when this approach is used in the land model, there is an adjustment to qss for stomatal and soil evap resistance. What is being done here is essentially like a ponded surface, but I wouldn't think that would produce unreasonable values (though this could be happening over hot soils where the qss could get very large). In the land model, LH and SH are constrained by available energy, which we don't have here (neither does sfc_ocean), also we are considering them here to be independent, which they also aren't in the land model. But again, none of this should produce "very large" values, unless there is an initialization issue with some of the inputs.

Reply:
@barlage it seems that `qss = fpvs(tskin(i))` is returning very large value for me even if `tskin` is in the range. Maybe we need to use another function. The values like `1442` is produced with curent version. Any idea?

Reply:
@uturuncoglu the qss coming from the function is vapor pressure in Pascals so 1400 is not unreasonable, it's the equivalent of about 0.009 kg/kg specific humidity

Reply:
If you are saying both fluxes are large, then I'm wondering if it's something with the input `ch` values